### PR TITLE
Windows file path fix#1374

### PR DESCRIPTION
**Pull request by Budlee**

### CODE REVIEW

File: spring-cloud-contract-stub-runner/src/main/java/org/springframework/cloud/contract/stubrunner/ContractDownloader.java [hunk 1]
```diff
@@ -193,8 +193,8 @@ private String surroundWithSeparator(String string) {
 	}
 
 	private String wrapWithAntPattern(String path) {
-		String changedPath = path.replace(File.separator, "/");
-		return "**" + surroundWithSeparator(changedPath).replace(File.separator, "/")
+		String changedPath = surroundWithSeparator(path).replace(File.separator, "/");
```
Comment (Budlee) [inline spring-cloud-contract-stub-runner/src/main/java/org/springframework/cloud/contract/stubrunner/ContractDownloader.java:196]: The test wants:
**/a/b/c/d/**
what is generated before is **//a/b/c/d//**
This corrects that by adjusting the order when createding the AntPattern

File: spring-cloud-contract-stub-runner/src/main/java/org/springframework/cloud/contract/stubrunner/GitRepo.java [hunk 2]
```diff
@@ -290,13 +290,13 @@ private boolean containsBranch(Git git, String label,
 		return false;
 	}
 
-	private void deleteBaseDirIfExists() {
+	private void deleteBaseDirIfExists(String errorMessage) {
 		if (this.basedir.exists()) {
 			try {
-				FileUtils.delete(this.basedir, FileUtils.RECURSIVE);
+				org.apache.commons.io.FileUtils.deleteDirectory(this.basedir);
```
Comment (Budlee) [inline spring-cloud-contract-stub-runner/src/main/java/org/springframework/cloud/contract/stubrunner/GitRepo.java:296]: Changed the apache commons as in the jgit lib it does not like deleting the pack.idx file from .git 
Apachecommons deletes correctly as it is handling the windows file path issues.

File: spring-cloud-contract-stub-runner/src/main/java/org/springframework/cloud/contract/stubrunner/GitRepo.java [hunk 3]
```diff
@@ -236,7 +236,7 @@ private Ref checkoutBranch(File projectDir, String branch) throws GitAPIExceptio
 			return command.call();
 		}
 		catch (GitAPIException e) {
-			deleteBaseDirIfExists();
+			deleteBaseDirIfExists("Failed to delete base directory");
```
Comment (Budlee) [inline spring-cloud-contract-stub-runner/src/main/java/org/springframework/cloud/contract/stubrunner/GitRepo.java:239]: Added the error message as a parameter as this is used when cloning and when checking out the repo so was throwing a very confusing error

File: spring-cloud-contract-stub-runner/src/test/groovy/org/springframework/cloud/contract/stubrunner/AetherStubDownloaderSpec.groovy [hunk 4]
```diff
@@ -58,7 +58,7 @@ class AetherStubDownloaderSpec extends Specification {
 		and:
 			String localRepo = AetherFactories.localRepositoryDirectory(true)
 			new File(localRepo, "org/springframework/cloud/spring-cloud-contract-spec"
-					.replaceAll("/", File.separator)).list().size() > 0
```
Comment (Budlee) [inline spring-cloud-contract-stub-runner/src/test/groovy/org/springframework/cloud/contract/stubrunner/AetherStubDownloaderSpec.groovy:61]: replaceAll was doing a regex pattern match and not a literal so messed up the test

File: spring-cloud-contract-stub-runner/src/test/java/org/springframework/cloud/contract/stubrunner/GitStubDownloaderTests.java [hunk 5]
```diff
@@ -86,12 +86,13 @@ public void should_return_a_null_downloader_for_a_non_git_repo() {
 	public void should_pick_stubs_for_group_and_artifact_with_version_from_a_git_repo()
 			throws Exception {
 		StubDownloaderBuilder stubDownloaderBuilder = new ScmStubDownloaderBuilder();
+		String contractFolderLocation = (file("/git_samples/contract-git/").getAbsolutePath()
```
Comment (Budlee) [inline spring-cloud-contract-stub-runner/src/test/java/org/springframework/cloud/contract/stubrunner/GitStubDownloaderTests.java:89]: in these tests made sure the location is all git://a/b/c/ else was resolving to git://C:\a\b\c/ and was messing up the clone

File: spring-cloud-contract-stub-runner/src/test/java/org/springframework/cloud/contract/stubrunner/FileStubDownloaderTests.java [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+
+package org.springframework.cloud.contract.stubrunner;
+
+import org.assertj.core.api.Assertions;
+import org.junit.Test;
+
+import org.springframework.core.io.Resource;
+
+/**
+ * @author Budlee
```
Comment (OlgaMaciaszek) [inline spring-cloud-contract-stub-runner/src/test/java/org/springframework/cloud/contract/stubrunner/FileStubDownloaderTests.java:25]: Please use your full name and surname in the `@author` clause.

**Comment by Budlee** [comment]
@OlgaMaciaszek 
This should be done now. Anything else needed?

File: spring-cloud-contract-stub-runner/src/test/java/org/springframework/cloud/contract/stubrunner/FileStubDownloaderTests.java (hunk 6 above)
Comment (Budlee) [reply spring-cloud-contract-stub-runner/src/test/java/org/springframework/cloud/contract/stubrunner/FileStubDownloaderTests.java]: REsolved

File: spring-cloud-contract-stub-runner/src/main/java/org/springframework/cloud/contract/stubrunner/FileStubDownloader.java [hunk 7]
```diff
@@ -20,12 +20,12 @@
 import java.io.IOException;
 import java.io.InputStream;
 import java.net.URI;
-import java.nio.file.Paths;
 import java.util.Collections;
 import java.util.List;
 import java.util.Map;
 import java.util.regex.Pattern;
 
+import org.apache.commons.io.FilenameUtils;
```
Comment (marcingrzejszczak) [inline spring-cloud-contract-stub-runner/src/main/java/org/springframework/cloud/contract/stubrunner/FileStubDownloader.java:28]: Do we need to add a new dependency just for this? I'd prefer not to introduce it.

Comment (Budlee) [reply spring-cloud-contract-stub-runner/src/main/java/org/springframework/cloud/contract/stubrunner/FileStubDownloader.java:28]: @marcingrzejszczak The issues I had was that on windows the org.eclipse.jgit.util.FileUtils; will not recursivly delete one of the files in .git and this this then leaves the tmp folder in tact. This fails the tests on Windows but passes on Unix.

Comment (marcingrzejszczak) [reply spring-cloud-contract-stub-runner/src/main/java/org/springframework/cloud/contract/stubrunner/FileStubDownloader.java:28]: So I'd prefer not to add a new dependency but write our own version of the deleting mechanism. There is the FileSystemUtils class in spring that can be helpful

Comment (Budlee) [reply spring-cloud-contract-stub-runner/src/main/java/org/springframework/cloud/contract/stubrunner/FileStubDownloader.java]: @marcingrzejszczak This is now removed. Using the Spring version that you suggested with works

**Comment by Budlee** [comment]
This should all be resolved. Any other feedback @marcingrzejszczak and  @OlgaMaciaszek

**Comment by marcingrzejszczak** [comment]
Great work @Budlee , thanks a lot!

**Pull request merged**